python/hail/api2/table.py

```diff
@@ -1244,3 +1277,420 @@ def format_type(typ):
+            only in the left table, the right table's fields will be missing.
+            For keys only in the right table, the left table's fields will be
+            missing.
+         - **left** -- Key present in the left table. For keys not found on
+            the right, the right table's fields will be missing.
+         - **right** -- Key present in the right table. For keys not found on
+            the right, the right table's fields will be missing.
+
+        Both tables must have the same number of keys and the corresponding
+        types of each key must be the same (order matters), but the key names
+        can be different. For example, if `table1` is keyed by fields ``['a',
+        'b']``, both of type ``Int32``, and `table2` is keyed by fields ``['c',
+        'd']``, both of type ``Int32``, then the two tables can be joined (their
+        rows will be joined where ``table1.a == table2.c`` and ``table1.b ==
+        table2.d``.
+
+        The key field names and order from the left table are preserved, while
+        the key fields from the right table are not present in the result.
+
+        Parameters
+        ----------
+        right : :class:`Table`
+            Table with which to join.
+        how : :obj:`str`
+            Join type. One of "inner", "outer", "left", "right".
+
+        Returns
+        -------
+        :class:`Table`
+            Joined table.
+        """
+
+        return Table(self._hc, self._jkt.join(right._jkt, how))
+
+    @handle_py4j
+    @typecheck_method(expr=BooleanExpression)
+    def forall(self, expr):
+        """Evaluate whether a boolean expression is true for all rows.
+
+        Examples
+        --------
+        Test whether `C1` is greater than 5 in all rows of the table:
+
+        >>> if table1.forall(table1.C1 == 5):
+        ...     print("All rows have C1 equal 5.")
+
+        Parameters
+        ----------
+        expr : :class:`hail.expr.expression.BooleanExpression`
+            Expression to test.
+
+        Returns
+        -------
+        :obj:`bool`
+        """
+        expr = to_expr(expr)
+        analyze(expr, self._row_indices, set(), set(self.columns))
+        base, cleanup = self._process_joins(expr)
+        if not isinstance(expr._type, TBoolean):
+            raise TypeError("method 'filter' expects an expression of type 'TBoolean', found {}"
+                            .format(expr._type.__class__))
+
+        return base._jkt.forall(expr._ast.to_hql())
+
+    @handle_py4j
+    @typecheck_method(expr=BooleanExpression)
+    def exists(self, expr):
+        """Evaluate whether a boolean expression is true for at least one row.
+
+        Examples
+        --------
+
+        Test whether `C1` is equal to 5 any row in any row of the table:
+
+        >>> if table1.exists(table1.C1 == 5):
+        ...     print("At least one row has C1 equal 5.")
+
+        :param str expr: Boolean expression.
+
+        :rtype: bool
```

Comment:
This should be
```
Parameters
----------
...
Returns
-------
...
```
no?

Reply:
ah, yep. I copied this and didn't edit the docs.